src/openrct2/world/Footpath.cpp

```diff
@@ -1716,24 +1716,17 @@ void footpath_update_path_wide_flags(const CoordsXY& footpathPos)
         // Spanned from 0x00F3EFA8 to 0x00F3EFC7 (8 elements) in the original
         TileElement* pathList[8];
 
-        // TODO: Use DirectionDelta
-        auto pathPos = footpathPos - CoordsXY{ COORDS_XY_STEP, COORDS_XY_STEP };
-        pathList[0] = footpath_can_be_wide(pathPos, height);
-        pathPos.y += COORDS_XY_STEP;
-        pathList[1] = footpath_can_be_wide(pathPos, height);
-        pathPos.y += COORDS_XY_STEP;
-        pathList[2] = footpath_can_be_wide(pathPos, height);
-        pathPos.x += COORDS_XY_STEP;
-        pathList[3] = footpath_can_be_wide(pathPos, height);
-        pathPos.x += COORDS_XY_STEP;
-        pathList[4] = footpath_can_be_wide(pathPos, height);
-        pathPos.y -= COORDS_XY_STEP;
-        pathList[5] = footpath_can_be_wide(pathPos, height);
-        pathPos.y -= COORDS_XY_STEP;
-        pathList[6] = footpath_can_be_wide(pathPos, height);
-        pathPos.x -= COORDS_XY_STEP;
-        pathList[7] = footpath_can_be_wide(pathPos, height);
-        pathPos.y += COORDS_XY_STEP;
+        auto pathPos = footpathPos + CoordsDirectionDelta[7];
+        Direction direction = 1;
+        int32_t i = 0;
+        while (i < 8)
+        {
+            pathList[i++] = footpath_can_be_wide(pathPos, height);
+            pathPos += CoordsDirectionDelta[direction];
+            pathList[i++] = footpath_can_be_wide(pathPos, height);
+            pathPos += CoordsDirectionDelta[direction];
+            direction = direction_next(direction);
+        }
```

Comment:
This likely works but we could do it simpler by having direction being 0->7. And the pos being a sum of the base footpathPos plus the Delta at the specified direction. It would be a different layout of the array though. Will need a little think to check if that is an issue.

Reply:
For reference, this is what the delta looks like at each position:
```
[0] = {-32, -32}
[1] = {-32, 0}
[2] = {-32, 32}
[3] = {0, 32}
[4] = {32, 32}
[5] = {32, 0}
[6] = {32, -32}
[7] = {0, -32}
```

Reply:
Which translates to:
```
[0] = footpathPos + CoordsDirectionDelta[7];
[1] = footpathPos + CoordsDirectionDelta[0];
[2] = footpathPos + CoordsDirectionDelta[4];
[3] = footpathPos + CoordsDirectionDelta[1];
[4] = footpathPos + CoordsDirectionDelta[5];
[5] = footpathPos + CoordsDirectionDelta[2];
[6] = footpathPos + CoordsDirectionDelta[6];
[7] = footpathPos + CoordsDirectionDelta[3];
```

Reply:
I've simplified it on the latest commit using the conversions above, please double check @duncanspumpkin .